app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt

```diff
@@ -17,22 +17,32 @@ class ExitProfileDialogFragment : DialogFragment() {
     // TODO(#1655): Re-restrict access to fields in tests post-Gradle.
     const val BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY =
       "BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY"
+    const val BOOL_IS_ADMINISTRATOR_CONTROLS_SELECTED_KEY =
+      "BOOL_IS_ADMINISTRATOR_CONTROLS_SELECTED_KEY"
+    const val INT_LAST_CHECKED_ITEM_KEY =
+      "INT_LAST_CHECKED_ITEM_KEY"
 
     /**
      * This function is responsible for displaying content in DialogFragment.
      *
      * @return [ExitProfileDialogFragment]: DialogFragment
      */
-    fun newInstance(isFromNavigationDrawer: Boolean): ExitProfileDialogFragment {
+    fun newInstance(
+      isFromNavigationDrawer: Boolean,
+      isAdministratorControlsSelected: Boolean,
```

Comment:
I'm assuming these both can't be true, so given that they're mutually exclusive I think we should model the API (both the function & the argument keys) to be based on an enum rather than booleans.

This might actually be a good opportunity to start implementing support for protos stored in bundles. I'll try to get a PR out later today that introduces a utility to do this easily. The idea would be to introduce a proto file + proto that represents the intent arguments for this fragment and then just populate a new object & save/retrieve it from the bundle. Will try to make the API work well for both intents & arguments.

This will be establishing a new pattern, but I think it's long-term the better way to go since it provides us with better flexibility & API design for fragments & activities.

Reply:
@BenHenning I think they both can be simultaneously true, the first one is true when we open Navigation Drawer and click `Switch Profile` (isFromNavigationDrawer = true) since it denotes that the dialogFragment created is initialized from the Navigation Drawer or not; in our case, it will always be true. 
The second denotes that Administrator Controls is checked or not, and it may be true or false independent of `isFromNavigationDrawer`.
So, we might not be able to use enum here.
>This might actually be a good opportunity to start implementing support for protos stored in bundles. I'll try to get a PR out later today that introduces a utility to do this easily. The idea would be to introduce a proto file + proto that represents the intent arguments for this fragment and then just populate a new object & save/retrieve it from the bundle. Will try to make the API work well for both intents & arguments.

>This will be establishing a new pattern, but I think it's long-term the better way to go since it provides us with better flexibility & API design for fragments & activities.

If it is a better way to design the API for intents and arguments, I'll definitely try to implement it, so will wait till you make a PR for this.